tide/core/network/MPINospin.cpp

```diff
@@ -53,6 +53,20 @@ namespace
 {
 const size_t nsec_start = 1000;
 const size_t nsec_max = 100000;
+
+int _waitForCompletion(MPI_Request req, MPI_Status* status)
+{
+    timespec ts{0, nsec_start};
+    int flag = 0;
+    while (!flag)
+    {
+        nanosleep(&ts, nullptr);
```

Comment:
Use C++? ```std::this_thread::sleep(std::chrono::nanoseconds(nsec_start));```

Reply:
OK can do, thanks for the suggestion although in this case C++ does not look simpler :-) This was a code sample I found somewhere some time ago and I didn't try to adapt it more than what was strictly necessary...

Reply:
Of course C++ is not shorter/simpler, but it's not C anymore :)

Reply:
done (without bugs I hope :-))